Can't compile v0.10.0 in the `no_std` environment

The following code compiles with the `derive_builder` crate v0.9.0:

```rust
#![no_std]

#[macro_use]
extern crate derive_builder;

extern crate alloc;

#[derive(Builder)]
#[builder(no_std)]
struct Foo {
    bar: i32,
}
```

However, it can't compile with v0.10.0, even without the default features, emitting the following errors:

```
   Compiling foo v0.1.0 (/tmp/tmp.obTGIdHacx/foo)
error[E0433]: failed to resolve: use of undeclared crate or module `std`
 --> src/lib.rs:8:10
  |
8 | #[derive(Builder)]
  |          ^^^^^^^ use of undeclared crate or module `std`
  |
  = note: this error originates in a derive macro (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0412]: cannot find type `String` in this scope
 --> src/lib.rs:8:10
  |
8 | #[derive(Builder)]
  |          ^^^^^^^ not found in this scope
  |
  = note: this error originates in a derive macro (in Nightly builds, run with -Z macro-backtrace for more info)
help: consider importing this struct
  |
9 | use alloc::string::String;
  |

error: aborting due to 2 previous errors

Some errors have detailed explanations: E0412, E0433.
For more information about an error, try `rustc --explain E0412`.
error: could not compile `foo`

To learn more, run the command again with --verbose.
```

Something depends on the `std` crate.

--- a/.github/workflows/build.yml
+++ b/.github/workflows/build.yml
@@ -49,3 +49,9 @@ jobs:
           cargo build ${{ matrix.features }} --verbose
           cargo test --no-fail-fast ${{ matrix.features }} --verbose -- --nocapture
           cargo doc
+
+      - name: no_std tests crate
+        run: |
+          cd derive_builder_no_std_tests
+          cargo build ${{ matrix.features }} --verbose
+          cargo test --no-fail-fast ${{ matrix.features }} --verbose -- --nocapture

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -1,2 +1,2 @@
 [workspace]
-members = ["derive_builder", "derive_builder_macro", "derive_builder_core"]
+members = ["derive_builder", "derive_builder_macro", "derive_builder_core", "derive_builder_no_std_tests"]

--- a/derive_builder/src/lib.rs
+++ b/derive_builder/src/lib.rs
@@ -541,6 +541,9 @@
 #![deny(warnings)]
 #![cfg_attr(not(feature = "std"), no_std)]
 
+#[cfg(not(feature = "std"))]
+extern crate alloc;
+
 extern crate derive_builder_core;
 extern crate derive_builder_macro;
 
@@ -552,6 +555,8 @@ pub use derive_builder_core::UninitializedFieldError;
 #[doc(hidden)]
 pub mod export {
     pub mod core {
+        #[cfg(not(feature = "std"))]
+        pub use alloc::string;
         #[cfg(not(feature = "std"))]
         pub use core::*;
         #[cfg(feature = "std")]

--- a/derive_builder_core/src/block.rs
+++ b/derive_builder_core/src/block.rs
@@ -67,7 +67,7 @@ mod test {
     use super::*;
 
     #[test]
-    #[should_panic(expected = r#"LexError"#)]
+    #[should_panic(expected = r#"lex error"#)]
     fn block_invalid_token_trees() {
         Block::from_str("let x = 2; { x+1").unwrap();
     }

--- a/derive_builder_core/src/builder.rs
+++ b/derive_builder_core/src/builder.rs
@@ -48,7 +48,7 @@ use Setter;
 ///     /// Uninitialized field
 ///     UninitializedField(&'static str),
 ///     /// Custom validation error
-///     ValidationError(String),
+///     ValidationError(::derive_builder::export::core::string::String),
 /// }
 ///
 /// impl ::derive_builder::export::core::convert::From<&'static str> for FooBuilderError {
@@ -57,8 +57,8 @@ use Setter;
 ///     }
 /// }
 ///
-/// impl ::derive_builder::export::core::convert::From<String> for FooBuilderError {
-///     fn from(s: String) -> Self {
+/// impl ::derive_builder::export::core::convert::From<::derive_builder::export::core::string::String> for FooBuilderError {
+///     fn from(s: ::derive_builder::export::core::string::String) -> Self {
 ///         Self::ValidationError(s)
 ///     }
 /// }
@@ -139,6 +139,8 @@ pub struct Builder<'a> {
     pub doc_comment: Option<syn::Attribute>,
     /// Emit deprecation notes to the user.
     pub deprecation_notes: DeprecationNotes,
+    /// Whether or not a libstd is used.
+    pub std: bool,
 }
 
 impl<'a> ToTokens for Builder<'a> {
@@ -219,7 +221,7 @@ impl<'a> ToTokens for Builder<'a> {
                         /// Uninitialized field
                         UninitializedField(&'static str),
                         /// Custom validation error
-                        ValidationError(String),
+                        ValidationError(::derive_builder::export::core::string::String),
                     }
 
                     impl ::derive_builder::export::core::convert::From<::derive_builder::UninitializedFieldError> for #builder_error_ident {
@@ -228,8 +230,8 @@ impl<'a> ToTokens for Builder<'a> {
                         }
                     }
 
-                    impl ::derive_builder::export::core::convert::From<String> for #builder_error_ident {
-                        fn from(s: String) -> Self {
+                    impl ::derive_builder::export::core::convert::From<::derive_builder::export::core::string::String> for #builder_error_ident {
+                        fn from(s: ::derive_builder::export::core::string::String) -> Self {
                             Self::ValidationError(s)
                         }
                     }
@@ -242,10 +244,13 @@ impl<'a> ToTokens for Builder<'a> {
                             }
                         }
                     }
+                ));
 
-                    #[cfg(not(no_std))]
-                    impl std::error::Error for #builder_error_ident {}
-            ));
+                if self.std {
+                    tokens.append_all(quote!(
+                        impl std::error::Error for #builder_error_ident {}
+                    ));
+                }
             }
         }
     }
@@ -327,6 +332,7 @@ macro_rules! default_builder {
             must_derive_clone: true,
             doc_comment: None,
             deprecation_notes: DeprecationNotes::default(),
+            std: true,
         }
     };
 }
@@ -346,7 +352,7 @@ mod tests {
                 /// Uninitialized field
                 UninitializedField(&'static str),
                 /// Custom validation error
-                ValidationError(String),
+                ValidationError(::derive_builder::export::core::string::String),
             }
 
             impl ::derive_builder::export::core::convert::From<::derive_builder::UninitializedFieldError> for FooBuilderError {
@@ -355,8 +361,8 @@ mod tests {
                 }
             }
 
-            impl ::derive_builder::export::core::convert::From<String> for FooBuilderError {
-                fn from(s: String) -> Self {
+            impl ::derive_builder::export::core::convert::From<::derive_builder::export::core::string::String> for FooBuilderError {
+                fn from(s: ::derive_builder::export::core::string::String) -> Self {
                     Self::ValidationError(s)
                 }
             }
@@ -370,7 +376,6 @@ mod tests {
                 }
             }
 
-            #[cfg(not(no_std))]
             impl std::error::Error for FooBuilderError {}
         ));
     }

--- a/derive_builder_core/src/macro_options/darling_opts.rs
+++ b/derive_builder_core/src/macro_options/darling_opts.rs
@@ -401,6 +401,10 @@ impl Options {
             must_derive_clone: self.requires_clone(),
             doc_comment: None,
             deprecation_notes: Default::default(),
+            std: {
+                let no_std: bool = self.no_std.into();
+                !no_std
+            },
         }
     }
 

--- a/derive_builder_no_std_tests/Cargo.toml
+++ b/derive_builder_no_std_tests/Cargo.toml
@@ -1,0 +1,11 @@
+[package]
+name = "derive_builder_no_std_tests"
+version = "0.1.0"
+authors = ["Andrey Kutejko <andy128k@gmail.com>"]
+edition = "2018"
+
+[features]
+clippy = []
+
+[dependencies]
+derive_builder = { path = "../derive_builder", default-features = false }

--- a/derive_builder_no_std_tests/src/main.rs
+++ b/derive_builder_no_std_tests/src/main.rs
@@ -1,0 +1,18 @@
+#![no_std]
+#![allow(unused)]
+
+#[macro_use]
+extern crate derive_builder;
+
+extern crate alloc;
+
+#[derive(Builder)]
+#[builder(no_std)]
+struct Foo {
+    bar: i32,
+}
+
+fn main() {
+    let foo = FooBuilder::default().build();
+    assert!(foo.is_err());
+}

